pandas/core/dtypes/inference.py

```diff
@@ -422,3 +422,38 @@ def is_sequence(obj) -> bool:
         return not isinstance(obj, (str, bytes))
     except (TypeError, AttributeError):
         return False
+
+
+def is_dataclass_instance(item):
+    """
+    Checks if the object is a data-class instance
+
+    Parameters
+    ----------
+    item : object
+
+    Returns
```

Comment:
add a versionadded tag

Reply:
not sure what you mean

Reply:
If you grep for versionversionadded you will see this in quite a few places. Here's an example:

https://github.com/pandas-dev/pandas/blob/8ff2ebd9b2cfbe4a3e1eb7893dc4343c01ad55b4/pandas/core/indexes/base.py#L1054

Should target 1.1.0 at this point

Reply:
can you update